esmvalcore/_main.py

```diff
@@ -99,7 +99,7 @@ def process_recipe(recipe_file, config_user):
     # parse recipe
     recipe = read_recipe_file(recipe_file, config_user)
     logger.debug("Recipe summary:\n%s", recipe)
-
+    recipe.write_filled_recipe()
```

Comment:
Maybe it would be better to include this in `recipe.run` (same goes for `recipe.write_html_summary`), because having to make all these separate calls to run a recipe makes it more complicated to use.